inbm/dispatcher-agent/dispatcher/workload_orchestration_2.py

```diff
@@ -0,0 +1,296 @@
+                    elif status_code == CSL_POLL_CMD_STATUS_CODE:
+                        self._process_online_mode_ok_status_result(return_json)
+        except DispatcherException as e:
+            self._dispatcher_broker.telemetry(
+                "Failure in switching Device workload Orchestration status to Online mode: {}".format(str(e)))
+
+    def _process_online_mode_ok_status_result(self, return_json: Dict) -> None:
+        """Checks if the device is in maintenance mode and if it does it calls the API to switch it to online mode.
+        The result json determines if the device orchestration status is changed to online mode or not.
+
+        @param return_json: Dict of device status mode and workload details
+        """
+        if return_json["Enabled"]:
+            self._dispatcher_broker.telemetry(
+                'Switching Device Workload Orchestration status to Online mode.')
+            (return_json, status_code) = self.switch_wo_status("false")
+            if status_code != CSL_CMD_STATUS_CODE:
+                self._dispatcher_broker.telemetry(
+                    'Failure in switching Device Workload Orchestration status to Online mode')
+            elif status_code == CSL_CMD_STATUS_CODE:
+                self._dispatcher_broker.telemetry(
+                    'Switched Device Workload Orchestration status to Online mode')
+
+    def _get_workload_orchestration_file_content(self, file: str) -> str:
+        """Reads and returns the workload orchestration file contents
+
+        @return: contents of the file
+        """
+        try:
+            with open(file) as file_content:
+                return file_content.read()
+        except OSError as e:
+            raise DispatcherException(
+                f"Could not load workload orchestration config file with error: {e}")
+
+    def get_wo_details(self) -> Tuple[Optional[str], Optional[str]]:
+        """Returns workload orchestration details
+
+        @return: device workload orchestration ip and token values
+        """
+        token = self.get_orchestrator_value(TOKEN)
+        ip_port = self.get_orchestrator_value(IP)
+        return token, ip_port
+
+    def poll_wo_status(self) -> Tuple[Dict, int]:
+        """Create and process the REST API call to poll device workload orchestration status
+
+        @return: API call return status code and json structure 
+        """
+        (token, ip_port) = self.get_wo_details()
+
+        if token is None or ip_port is None:
+            raise DispatcherException(" Workload-Orchestration IP and Token details Not Found")
+        try:
+            url = self._get_workload_orchestration_file_content(ip_port) + '/api/v1/nodes/' \
+                + WorkloadOrchestration.get_hostname() \
+                + '/maintenance?token=' + self._get_workload_orchestration_file_content(token)
+
+            result = requests.get(url, verify=self.get_orchestrator_value(CSL_CA))
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Refactoring$</h2>**
**Suggestion:** Refactor the repeated code for making REST API calls into a separate method to improve code reusability and maintainability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            def make_api_call(self, method, endpoint, data=None):
                base_url = self._get_workload_orchestration_file_content(self.get_orchestrator_value(IP))
                token = self._get_workload_orchestration_file_content(self.get_orchestrator_value(TOKEN))
                url = f"{base_url}{endpoint}?token={token}"
                headers = {"Content-Type": "application/json"} if data else {}
                verify = self.get_orchestrator_value(CSL_CA)
                if method == 'GET':
                    result = requests.get(url, verify=verify)
                elif method == 'PATCH':
                    result = requests.patch(url, data=json.dumps(data), headers=headers, verify=verify)
                return result.json(), result.status_code
```